### PR TITLE
Migrate centralized sampling tests from aws-otel-test-framework

### PR DESCRIPTION
**Pull request by jj22ee**
Description:

Integration tests through a Sample Java App to test centralized sampling via OTel Java are currently defined in the [aws-observability/aws-otel-test-framework](https://github.com/aws-observability/aws-otel-test-framework/tree/terraform/centralized-sampling-tests) repository. The goal is to migrate the sample app along with its integration tests into this `aws-observability/aws-otel-community` repository.

Modifications have been made to the sample app's gradle files and workflow file to work in this repository.

The GitHub secret key `AWS_CENTRALIZED_SAMPLING_ROLE` needs to be populated in this repository's settings before this PR is merged.

Link to tracking Issue: N/A

Testing Performed:
Workflow tested in a fork of this repository - https://github.com/jj22ee/aws-otel-community/actions/runs/4432041075

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/centralized-sampling-tests.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+permissions:
+  id-token: write
+
+jobs:
+  run_tests:
+    name: Run centralized sampling integration tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
```
Comment (bryan-aguilar) [inline .github/workflows/centralized-sampling-tests.yml:27]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/centralized-sampling-tests.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+      - uses: actions/setup-java@v1
```
Comment (bryan-aguilar) [inline .github/workflows/centralized-sampling-tests.yml:29]:
```suggestion
      - uses: actions/setup-java@v3
```

File: .github/workflows/centralized-sampling-tests.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        with:
+          java-version: 17
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
```
Comment (bryan-aguilar) [inline .github/workflows/centralized-sampling-tests.yml:33]:
```suggestion
        uses: aws-actions/configure-aws-credentials@v1-node16
```

**Comment by jj22ee** [comment]
Added the updates, but the [workflow fails](https://github.com/jj22ee/aws-otel-community/actions/runs/4514098780) now.
I'll need to update this.

**Review by bryan-aguilar — APPROVED** [review]

**Comment by bryan-aguilar** [comment]
Can you please update the `.github/dependabot.yml` file to add the new folders and dependencies.

File: .github/dependabot.yml [hunk 4]
```diff
@@ -34,3 +34,13 @@ updates:
...
+  - package-ecosystem: "gradle"
+    directory: "/centralized-sampling-tests/integration-tests"
+    schedule:
+      interval: "weekly"
+      day: "sunday"
+  - package-ecosystem: "gradle"
+    directory: "/centralized-sampling-tests/sample-apps/spring-boot"
+    schedule:
+      interval: "weekly"
+      day: "sunday"
```
Comment (bryan-aguilar) [inline .github/dependabot.yml:46]:
```suggestion
      day: "sunday"

```

**Pull request merged**